lib/rubocop/cop/sorbet/redundant_extend_t_sig.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Forbids the use of redundant `extend T::Sig`. Meant for use in
+      # applications that monkey patch `Module.include(T::Sig)` globally,
+      # which would make it redundant.
+      #
+      # @safety
+      #   This cop should not be enabled in applications that have not monkey
+      #   patched `Module`.
+      #
+      # @example
+      #   # bad
+      #   class Greeter
+      #     extend T::Sig
+      #     sig { void }
+      #     def say_hello = puts "hello"
+      #   end
+      #
+      #   # good
+      #   class Greeter
+      #     sig { void }
+      #     def say_hello = puts "hello"
+      #   end
+      #
+      class RedundantExtendTSig < Base
+        extend AutoCorrector
```

Comment:
Looks like the way we generate docs doesn't support the `Base` API, and requires the old `Cop` API.  I've updated it to the legacy approach to correction, and will open an issue to migrate.